src/manifolds/FixedRankMatrices.jl

```diff
@@ -408,3 +421,34 @@ function zero_tangent_vector!(
     X.Vt .= zeros(eltype(X.Vt), k, n)
     return X
 end
+
+@doc raw"""
+    embed(p::SVDMPoint)
+
+Embed the point `p` from its `SVDMPoint` representation.
+"""
+
+function embed!(::FixedRankMatrices, q, p)
+    return q .= p.U * Diagonal(p.S) * p.Vt
+end
+function embed(M::FixedRankMatrices{m,n,k}, p) where {m,n,k}
```

Comment:
Do we need this method defined, or is the default `embed` in ManifoldsBase sufficient?
https://github.com/JuliaManifolds/ManifoldsBase.jl/blob/v0.10.4/src/ManifoldsBase.jl#L320-L324

Reply:
For some reason that I don't understand `embed` throws an error if I don't implement it. But kellertuer said that `embed!` should be enough, so I don't know.
```julia
MethodError: no method matching mul!(::SVDMPoint{Array{Float64,2},Array{Float64,1},Array{Float64,2}}, ::Array{Float64,2}, ::Array{Float64,2}, ::Bool, ::Bool)
Closest candidates are:
  mul!(!Matched::StridedArray{T, 2}, ::StridedVecOrMat{T}, ::StridedVecOrMat{T}, ::Number, ::Number) where T<:Union{Complex{Float32}, Complex{Float64}, Float32, Float64} at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/matmul.jl:167
  mul!(!Matched::AbstractArray{T,2} where T, ::Union{AbstractArray{T,1}, AbstractArray{T,2}} where T, ::Union{AbstractArray{T,1}, AbstractArray{T,2}} where T, ::Number, ::Number) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/matmul.jl:235
  mul!(!Matched::AbstractArray, !Matched::Number, ::AbstractArray, ::Number, ::Number) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/generic.jl:124
  ...

Stacktrace:
 [1] mul! at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/matmul.jl:208 [inlined]
 [2] embed!(::FixedRankMatrices{50,50,10,ℝ}, ::SVDMPoint{Array{Float64,2},Array{Float64,1},Array{Float64,2}}, ::SVDMPoint{Array{Float64,2},Array{Float64,1},Array{Float64,2}}) at /Users/quentinrebjock/.julia/dev/Manifolds/src/manifolds/FixedRankMatrices.jl:437
 [3] embed(::FixedRankMatrices{50,50,10,ℝ}, ::SVDMPoint{Array{Float64,2},Array{Float64,1},Array{Float64,2}}) at /Users/quentinrebjock/.julia/dev/ManifoldsBase/src/ManifoldsBase.jl:322
 [4] top-level scope at In[3]:1
 [5] include_string(::Function, ::Module, ::String, ::String) at ./loading.jl:1091
```

I experimented a little and I think it's because `embed` doesn't allocate the right type for `q`. And I think something should be added to the manifold `FixedRankMatrices` so that `embed` recognizes the right type to allocate. But I haven't figured out what to do.

Reply:
That's interesting. Yeah, looks like a bug that should be handled elsewhere first. @kellertuer any ideas what's going wrong here?

Reply:
Yes, we are missing the correct allocation function https://juliamanifolds.github.io/Manifolds.jl/latest/interface.html#Allocation, which is an advanced variant of `similar`basically, since for example the log has two points as input but has to allocate the right tangent vector as result memory. I know how that works and how to do it, I just have a meeting today first.